pkg/performanceprofile/controller/performanceprofile/hypershift/hypershift_test.go

```diff
@@ -249,3 +251,50 @@ spec:
 		})
 	}
 }
+
+func TestDecodeManifest(t *testing.T) {
+	testCases := []struct {
+		name     string
+		manifest []byte
+		into     runtime.Object
+		isLoaded bool
+	}{
+		{
+			name:     "decode kubelet config into machine config",
+			manifest: []byte(kubeletConfig1),
+			into:     &machineconfigv1.MachineConfig{},
+			isLoaded: false,
+		},
+		{
+			name:     "decode machine config into machine config",
+			manifest: []byte(machineConfig1),
+			into:     &machineconfigv1.MachineConfig{},
+			isLoaded: true,
+		},
+		{
+			name:     "decode performance profile into performance profile",
+			manifest: []byte(perfprofOne),
+			into:     &performancev2.PerformanceProfile{},
+			isLoaded: true,
+		},
+	}
+
+	if err := performancev2.AddToScheme(scheme.Scheme); err != nil {
+		t.Fatal(err)
+	}
+	if err := machineconfigv1.AddToScheme(scheme.Scheme); err != nil {
+		t.Fatal(err)
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ok, err := DecodeManifest(tc.manifest, scheme.Scheme, tc.into)
+			if err != nil {
+				t.Errorf("failed to decode manifest into %T: %v", tc.into, err)
+			}
```

Comment:
should we add a test which fails, such as we expect a non-nil err?

Reply:
Added

Reply:
I would have expected a `expectedError` or something similar, but this is not blocking

Reply:
if `err != nil` it means the decoding itself failed. 
The case we care about is whether the manifest which we decoded matches the provided object.
This check is done in the `!ok && tc.isLoaded`  below

Reply:
ok, makes sense. Thanks!